tests/test_lag_2.py

```diff
@@ -106,21 +133,88 @@ def verify_lag_lacp_timing(ptfhost, vm_name, lacp_timer, exp_iface):
+        lag_facts = duthost.lag_facts(host = duthost.hostname)['ansible_facts']['lag_facts']
+
+        # Verify lag member is marked deselected for the shutdown porta and all other lag member interfaces are marked selected
+        for po_intf in po_interfaces.keys():
+            if po_intf != intf:
+                assert lag_facts['lags'][lag_name]['po_stats']['ports'][po_intf]['runner']['selected']
+            else:
+                assert not lag_facts['lags'][lag_name]['po_stats']['ports'][po_intf]['runner']['selected']
+
+        # Verify PortChannel's interface are marked down/up correctly if it should down/up
+        if po_flap == True:
+            assert lag_facts['lags'][lag_name]['po_intf_stat'] == 'Down'
+        else:
+            assert lag_facts['lags'][lag_name]['po_intf_stat'] == 'Up'
+    finally:
+        # Bring back port in case test error and left testbed in unknow stage
+        # Bring up neighbor interface
+        set_neighbor_interface(vm_host, neighbor_interface, shut=False)
+
+        # Verify PortChannel interfaces are up correctly
+        for po_intf in po_interfaces.keys():
+            if po_intf != intf:
+                command = 'bash -c "teamdctl %s state dump" | python -c "import sys, json; print json.load(sys.stdin)[\'ports\'][\'%s\'][\'link\'][\'up\']"' % (lag_name, po_intf)
+                wait_until(wait_timeout, delay, check_shell_output, duthost, command)
+
+        # Refresh lag facts
+        lag_facts = duthost.lag_facts(host = duthost.hostname)['ansible_facts']['lag_facts']
+        for po_intf in po_interfaces.keys():
+            assert lag_facts['lags'][lag_name]['po_stats']['ports'][po_intf]['runner']['selected'] == True
+        
+        assert lag_facts['lags'][lag_name]['po_intf_stat'] == 'Up'
+
+def get_lag_intfs(lag_facts, lag_name):
+    # Figure out interface informations
+    po_interfaces = lag_facts['lags'][lag_name]['po_config']['ports']
+    intf = lag_facts['lags'][lag_name]['po_config']['ports'].keys()[0]
+    return intf, po_interfaces
 
-    base_path = os.path.dirname(os.path.realpath(__file__))
-    lab_conn_graph_file = os.path.join(base_path, "../ansible/files/lab_connection_graph.xml")
-    result = localhost.conn_graph_facts(host=host, filename=lab_conn_graph_file)['ansible_facts']
-    return result
+def check_flap(lag_facts, lag_name):
+    po_intf_num = len(lag_facts['lags'][lag_name]['po_config']['ports'])
+    po_min_links = lag_facts['lags'][lag_name]['po_config']['runner']['min_ports']
+    return ((po_intf_num - 1) * 100 / po_min_links) < 75
 
 def set_interface_lacp_rate(vm_host, intf, mode):
     vm_host.eos_config(
         lines=['lacp rate %s' % mode],
         parents='interface %s' % intf)
-    logging.info("Set interface [%s] lacp rate to [%s]" % (intf, mode))
\ No newline at end of file
+    logging.info("Set interface [%s] lacp rate to [%s]" % (intf, mode))
+
+def set_neighbor_interface(vm_host, neighbor_interface, shut):
```

Comment:
Does it make sense to make this a method of vm_host?

Reply:
Yes. I can move this to EosHost in next PR